components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIAdminImpl.java

```diff
@@ -81,10 +84,31 @@ public void deleteLabel(String labelId) throws APIManagementException{
      * @param label             content to update
      * @throws APIManagementException if failed to update label
      */
-    public Label updateLabel(Label label) throws APIManagementException{
+    public Label updateLabel(String tenantDomain, Label label) throws APIManagementException {
+        if (isLableNameExists(tenantDomain, label)) {
+            APIUtil.handleException("Label with name " + label.getName() + " already exists");
+        }
         return apiMgtDAO.updateLabel(label);
     }
 
+    /**
+     *
+     * @param label content to check
+     * @return whether label is already added or not
+     * @throws APIManagementException
+     */
+    public boolean isLableNameExists(String tenantDomain, Label label) throws APIManagementException {
+        List<Label> ExistingLables = apiMgtDAO.getAllLabels(tenantDomain);
+        if (!ExistingLables.isEmpty()) {
```

Comment:
checking list size is redundant here.  Loop continues only if it is not empty.

Reply:
fixed in a079997